File: Algorithms(Python)/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice=prices[0]
        maxProfit=0
        for i in range(len(prices)):
            if prices[i]<minPrice:
                minPrice=prices[i]
            else:
                if maxProfit<prices[i]-minPrice:
                    maxProfit=prices[i]-minPrice
        return maxProfit

File: Algorithms(Java)/Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit=0;
        int minPrice=prices[0];
        for(int i=1;i<prices.length;i++){
            if (prices[i]<minPrice){
                minPrice=prices[i];
            }else{
                if (maxProfit<prices[i]-minPrice){
                    maxProfit=prices[i]-minPrice;
                }
            }
        }
        return maxProfit;
    }
}